update-tags.mjs

```diff
@@ -0,0 +1,47 @@
+import { readFile, writeFile } from 'fs/promises'
+import { exec } from 'child_process'
+import { glob as baseGlob } from 'glob'
+import packageJSON from './package.json' assert { type: 'json' }
+
+const tagPath = `/v${packageJSON.version}/`
+
+async function glob(path) {
+  return await baseGlob(path, {
+    ignore: ['test/**', 'node_modules/**'],
+  })
+}
+async function loadAndUpdateFiles(oldValue, newValue) {
+  const schemaFilenames = await glob('**/*.schema.json')
+  await Promise.all(
+    schemaFilenames.map(async (match) => {
+      let schema = JSON.parse(
+        await readFile(match, {
+          encoding: 'utf-8',
+        })
+      )
+
+      schema['$id'] = schema['$id'].replace(oldValue, newValue)
+      await writeFile(match, JSON.stringify(schema, null, 2))
+    })
+  )
+  const exampleFilenames = await glob('**/*.example.json')
+  await Promise.all(
+    exampleFilenames.map(async (path) => {
+      const example = JSON.parse(
+        await readFile(path, {
+          encoding: 'utf-8',
+        })
+      )
+      example['$schema'] = example['$schema'].replace(oldValue, newValue)
+      await writeFile(path, JSON.stringify(example, null, 2))
+    })
+  )
+}
+
+if (process.argv.some((x) => x == '--reset')) {
+  await loadAndUpdateFiles(tagPath, '/main/')
+} else {
+  await loadAndUpdateFiles('/main/', tagPath)
+}
+
+exec('npx prettier -w .')
```

Comment:
Instead of running prettier, can we pass arguments to `JSON.stringify` that ensure that it will match Prettier's settings?

Reply:
Mostly, though it seems like the one specific that is getting in the way is array formatting:

json.stringify will output an array like this:

```
   "enum": [
        "circle",
        "ellipse",
        "rectangle"
      ],
```

where prettier parses it as short enough for one line:

```
"enum": ["circle", "ellipse", "rectangle"],
```

Reply:
OK. Instead of calling it from the CLI, would you please use the API? https://prettier.io/docs/en/api.html